### PR TITLE
Edge Agent: Fix bug where Edge Agent is updated without backing image

### PR DESCRIPTION
**Pull request by and-rewsmith**
I was doing some manual validation of my the Edge Agent feature I contributed in a prior PR:
https://github.com/Azure/iotedge/pull/6508

This validation turned up a bug relating to some side effects of how we restructured the command factories:
1. Update commands no longer are a GroupCommand with a PrepareUpdateCommand nested inside. This meant that on EdgeAgent update, new EdgeAgent image was not pulled before EdgeAgent went down.
2. Related to above, EdgeAgent needs dedicated PrepareUpdate command that will throw PrerequisiteException to block subsequent plan execution commands from running.

I have fixed both of these in this PR and added a test for when EdgeAgent container is updated with this feature flag on.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Review by ancaantochi — APPROVED** [review]

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json [hunk 1]
```diff
@@ -395,5 +399,162 @@
...
+          "startupOrder": 0,
+          "status": "running",
+          "restartPolicy": "always"
+        }
+      },
+      "modules": {
+        "tempSensor": {
+          "type": "docker",
+          "settings": {
+            "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-simulated-temperature-sensor:20191106.7-linux-amd64",
```
Comment (damonbarry) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json:444]: Replace hard-coded ACR image ref with MCR-published image ref, here and 3 more instances below

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json [hunk 2]
```diff
@@ -395,5 +399,162 @@
...
+      "modules": {
+        "edgeAgent": {
+          "exitCode": 0,
+          "runtimeStatus": "running",
+          "status": "running",
+          "restartPolicy": "always",
+          "startupOrder": 2,
+          "type": "docker",
+          "settings": {
+            "image": "mcr.microsoft.com/azureiotedge-agent:2.0",
```
Comment (damonbarry) [inline edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json:472]: 2.0 doesn't exist...is this intentional?

**Review by damonbarry — CHANGES_REQUESTED** [review]
Thanks for catching this, and for including a test! One requested change, one update.

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json (hunk 2 above)
Comment (and-rewsmith) [reply edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json:472]: 2.0 is used above in an existing test that I didn't write. Image is never actually pulled, this integration test is testing the ordering that the commands are executed.

File: edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json (hunk 1 above)
Comment (and-rewsmith) [reply edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Integration.Test/test-configs/module-priority-test-config.json:444]: These are actually never run or pulled. I will still change it though.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**